CMakeLists.txt

```diff
@@ -209,6 +209,17 @@ if("${TREAT_WARNINGS_AS_ERRORS}")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
 endif()
 
+if("${GENERATE_PGO}")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-generate=PGO_profile")
+endif()
+
+if("${USE_PGO}")
+  # Wno-missing-profile is necesary since we won't have profiling data for all
+  # files from just these benchmarks
+  set(CMAKE_CXX_FLAGS
+      "${CMAKE_CXX_FLAGS} -fprofile-use=PGO_profile -Wno-missing-profile")
+endif()
```

Comment:
Technically these will only work with GCC out of the box. With Clang, one needs to take the intermediate files that profiling produces and aggregate them together. I started down that path, but was never able to get it to work. But decided that if we're adding this in order to try this out on CircleCI for a little bit, GCC only is ok (since AFAICT that's all that's being used in CircleCI). If this does help, and we want to add this for real we can add instructions for how to do this in Clang if that's helpful